basiclti/basiclti-portlet/src/java/org/sakaiproject/blti/ProviderServlet.java

```diff
@@ -262,13 +262,23 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 		}
 
 		Map payload = getPayloadAsMap(request);
-
+		
+	   if(isEmailTrustedAndTrustedConsumerBothEnabled()){
+		   M_log.warn("Both Email Trusted and Trusted Consumer property is enabled, this is invalid  IP=" + ipAddress);
+			response.sendError(HttpServletResponse.SC_FORBIDDEN,
+					"Both Email Trusted and Trusted Consumer property is enabled, this is invalid ");
+			return;
+	   }
+	    
+         
 
 		// Get the list of highly trusted consumers from sakai.properties.
 		// If the incoming consumer is highly trusted, we use the context_id and site_id as is,
 		// ie without prefixing them with the oauth_consumer_key first.
 		// We also don't both checking their roles in the site.
         boolean isTrustedConsumer = isTrustedConsumer(payload);
+        
+        boolean isEmailTrustedConsumer = isEmailTrustedConsumer(payload);
 
```

Comment:
I'd move the check for both settings set to here - it's more efficient and avoids duplicating constants):

if (isTrustedConsumer && isEmailTrustedConsumer)
{
M_log.warn()
return;
}